pkg/controllers/provisioning/provisioner.go

```diff
@@ -95,24 +91,17 @@ func (p *Provisioner) Builder(_ context.Context, mgr manager.Manager) controller
 }
 
 func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (result reconcile.Result, err error) {
+	// Batch pods
+	if triggered := p.batcher.Wait(ctx); !triggered {
+		return reconcile.Result{}, nil
+	}
 	// If the provisioning loop fails for any reason, retrigger it,
 	// since pod watch events have already been processed
 	defer func() {
 		if err != nil {
 			p.Trigger()
 		}
 	}()
-	// Batch pods
-	if triggered := p.batcher.Wait(ctx); !triggered {
-		return reconcile.Result{}, nil
-	}
-
-	// wait to ensure that our cluster state is synced with the current known nodes to prevent over-provisioning
-	if WaitForClusterSync {
```

Comment:
I'm trying to figure out if this or something like it still needed to handle startup? 

E.g. Karpenter launches and the cluster has 500 nodes, one of which is in the process of starting up (just launched and then Karpenter had crashed).  We'll now launch a new node if we haven't observed the state of the in-flight node.

Reply:
Discussed offline -- should be pretty minor, this could occur if a very large number of in flight nodes could not be reconciled in time after a crash/boot plus the batching window. Keeping it simple for now, as this is going to change soon with machine CR, anyways.